src/IdentityServer/Extensions/ValidatedAuthorizeRequestExtensions.cs

```diff
@@ -154,8 +154,8 @@ private static NameValueCollection ToOptimizedRawValues(this ValidatedAuthorizeR
                 // https://openid.net/specs/openid-connect-core-1_0.html#JWTRequests 
                 // requires client id and response type to always be in URL
                 if (key == OidcConstants.AuthorizeRequest.ClientId ||
-                    key == OidcConstants.AuthorizeRequest.ResponseType ||
-                    !request.RequestObjectValues.Any(x => x.Type == key))
+                    key == OidcConstants.AuthorizeRequest.ResponseType || 
+                    request.RequestObjectValues.All(x => x.Type != key))
```

Comment:
Both would have identical performance because both stop enumeration after the outcome can be determined - Any() on the first item the passed predicate evaluates to true and All() on the first item the predicate evaluates to false.
In this case, All is more readable

Reply:
If the list has 100 items, `All` must execute all of them, but `Any` can exit if the first one evals to true, no? That's why I thought the performance of `Any` was better. 

As for the readability, I don't agree on this one :)

Reply:
```csharp
public static bool Any<TSource>(this IEnumerable<TSource> source, 
                                Func<TSource, bool> predicate)
{
    if (source == null)
    {
        throw Error.ArgumentNull("source");
    }
    if (predicate == null)
    {
        throw Error.ArgumentNull("predicate");
    }
    foreach (TSource local in source)
    {
        if (predicate(local))
        {
            return true;
        }
    }
    return false;
}

public static bool All<TSource>(this IEnumerable<TSource> source, Func<TSource, bool> predicate)
{
    if (source == null)
    {
        throw Error.ArgumentNull("source");
    }
    if (predicate == null)
    {
        throw Error.ArgumentNull("predicate");
    }
    foreach (TSource current in source)
    {
        if (!predicate(current))
        {
            return false;
        }
    }
    return true;
}

```
If you think it is not readable.  I change the code.  The all section does not count all items

Reply:
Oh, yea... now that I think about it, it's just the opposite logic... so both can short circuit. Duh!